src/js/src/components/SearchSingleItemComponents/SearchSingleItemImages.vue

```diff
@@ -0,0 +1,105 @@
+<template>
+  <div v-if="imageSrcs.length > 0" class="singleEntryImages">
+    <div class="singleEntryImagesHeadline">
+      <p class="highlightText entryInfo imageHeadline">
+        Images:
+      </p>
+      <p> 
+        showing {{ showNumberOfPictures(imageSrcs.length) }} out of {{ imageSrcs.length }}
+      </p>
+      <button v-if="imageSrcs.length > 4" class="allImagesButton" @click="toggleAllImagesShown">
+        {{ shownImagesButtonText }}
+      </button> 
+    </div>
+    <div v-for="(item, index) in shownImages(imageSrcs)"
+         :key="index"
+         class="previewImageContainer">
+      <img
+
+        loading="lazy"
+        class="previewImage"
+        :src="item.imageUrl + '&height=200&width=200'"
+        @click="toggleFullImage(index)">
+      <search-single-item-focus-image v-if="showFullImage === index"
+                                      :image="item.downloadUrl"
+                                      :index="index"
+                                      @close-window="closeWindow" />
+    </div>
+  </div>
+</template>
+
+<script>
+//import { mapState, mapActions } from 'vuex'
+import { requestService } from '../../services/RequestService'
+import SearchSingleItemFocusImage from './SearchSingleItemFocusImage.vue'
+
+export default {
+  name: 'SearchSingleItemImages',
+  components: {  
+    SearchSingleItemFocusImage
+  },
+  props: {
+    offset: {
+      type: Number,
+      required: true
+    },
+    source: {
+      type: String,
+      required: true
+    },
+  },
+  data () {
+    return {     
+      imageSrcs:[],
+      allImagesShown:false,
+      shownImagesButtonText: 'See all images',
+      showFullImage:null
+    }
+  },
+  computed: {
+  },
+  watch: { 
+    source: function() { // watch it
```

Comment:
Out of curiosity - why du you need to watch the source here?

Reply:
Because Vue tries to reuse elements when we load in now items after a new search - And i need to watch at least one of the props, to make sure we update the images when a new result is added. I noticed that the results updated, but the images didn't, so new results had old images. This solves that problem.

Reply:
👍